### ISSUE TITLE
Feature request: Async with filtering

### ISSUE THREAD

**Issue by stivon** [kenwheeler/slick#515]
Hi,

We have 2 sliders on the same configuration as the "Slider as Nav For / sync" example.
When you try to filter those sliders, an issue is encountered, see example : 
http://jsfiddle.net/dk0nsf9s/2/
At the click the thumbnail 3, it's the big image 5 that appears.

Thanks you a lot your help

**Comment by kalenjohnson** [comment]
I've updated stivon's jsfiddle for 1.5.0: http://jsfiddle.net/vpa60kt9/2/

The interesting thing is that the dots work correctly, but it's when you click on a slide in the carousel nav that things get all funky.

**Comment by wjthomas9** [comment]
Yeah, I'm having this issue as well. We have 2 sliders, using the nav syncing. It seems that after using slickFilter, the two sliders don't quite work.

**Comment by wjthomas9** [comment]
So, it appears currentSlide and nextSlide are no longer accurate after a slickFilter. They are still 0-incremented, but they are no longer a reliable way to get the current and next slides by data-slick-index.

**Comment by JiveDig** [comment]
Any fixes? Looks similar to https://github.com/kenwheeler/slick/issues/2500, but I can't get anything to work via slickFilter.

**Comment by japherwocky** [comment]
On a current project, we've done something like this:

        $('#asNavForElement .slide').each( function(i) {
            $(this).attr('data-slick-index', i);  // .data() does NOT work
        })

after we apply the filtering to set the slick-index as the code seems to expect.

**Comment by patrickroswellstudios** [comment]
While japherwocky is more efficient when things are filtered once but clicked on repeatedly, fixing the library might be in
Slick.prototype.asNavFor = function(index) {
this:

             asNavFor.each(function() {
                 var target = $(this).slick('getSlick');
                 if(!target.unslicked) {
+                     if (_.$slidesCache !== null) {
+                          var filteredIndex = index;
+                          target.$slides.each(function(slideIndex,slideElement) {
+                               if (filteredIndex == $(slideElement).data('slick-index')) {
+                                    index = slideIndex;
+                               }
+                          )
+                     }
                     target.slideHandler(index, true);
                 }
             });
